backend/canisters/user_index/api/src/queries/external_achievements.rs

```diff
@@ -0,0 +1,32 @@
+use candid::CandidType;
+use ts_export::ts_export;
+use types::TimestampMillis;
+
+#[ts_export(user_index, external_achievements)]
+#[derive(CandidType, Debug)]
+pub struct Args {
+    pub updates_since: TimestampMillis,
+}
+
+#[ts_export(user_index, external_achievements)]
+#[derive(CandidType, Debug)]
+pub enum Response {
+    Success(SuccessResult),
+    SuccessNoUpdates,
+}
+
+#[ts_export(user_index, external_achievements)]
+#[derive(CandidType, Debug)]
+pub struct SuccessResult {
+    pub timestamp: TimestampMillis,
+    pub achievements_added: Vec<ExternalAchievement>,
+    pub achievements_removed: Vec<ExternalAchievement>,
+}
+
+#[ts_export(user_index, external_achievements)]
+#[derive(CandidType, Debug)]
+pub struct ExternalAchievement {
```

Comment:
Expiry date?
Because then the backend doesn't need to return any that have expired since the frontend will already know that.

Reply:
Simpler without